flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetInputFormat.java

```diff
@@ -160,13 +168,16 @@ public void setFilterPredicate(FilterPredicate filterPredicate) {
     public void open(FileInputSplit split) throws IOException {
         // reset the flag when open a new split
         this.skipThisSplit = false;
+        // TODO why is this needed to read the parquet schema from file in open() ? Can't we use the
+        //  already extracted schema (extracted in the constructor) ?
         org.apache.hadoop.conf.Configuration configuration =
                 new org.apache.hadoop.conf.Configuration();
         InputFile inputFile =
                 HadoopInputFile.fromPath(
                         new org.apache.hadoop.fs.Path(split.getPath().toUri()), configuration);
         ParquetReadOptions options = ParquetReadOptions.builder().build();
         ParquetFileReader fileReader = new ParquetFileReader(inputFile, options);
+
         MessageType fileSchema = fileReader.getFileMetaData().getSchema();
         MessageType readSchema = getReadSchema(fileSchema, split.getPath());
```

Comment:
Here we should check if `fieldNames` is null and fill them. If set, call `getReadSchema` as is.